### PR TITLE
docs/development: initial version

### PR DESCRIPTION
**Pull request by freenerd**
While working on https://github.com/mapbox/ecs-watchbot/pull/251 I realized that there are no releasing docs. So I wrote some.

Todo:

- [ ] 👀@jakepruitt 
- [ ] Improvements
- [ ] Merge

### CODE REVIEW

File: docs/development.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+```
+
+### Releasing a version
+
+Ideally you track the release of a new version in an issue with a checklist
+
+- [ ] Merge all PRs intended for the release
+- [ ] Determine the new version number `x.y.z` after [SEMVER](https://semver.org/)
+- [ ] Verify that the [changelog](../changelog.md) includes all relevant changes under the new version number
+- [ ] Manually change the version number in package.json
```
Comment (arunasank) [inline docs/development.md]: @freenerd any reason why we don't want to recommend an `nom semver <patch|minor|major>`, which would auto generate a tag for us too?

Comment (jakepruitt) [reply docs/development.md]: ☝️ seconded, I think `npm version <patch|minor|major>` relieves the headache of manually changing the package and creating a git tag.

**Comment by rclark** [comment]
`feature request`: this doc should include information about how, when, and whether your commits will build binaries.

**Comment by freenerd** [comment]
@arunasank @jakepruitt didn't know about `npm version`. updated docs and package.json but didn't test it yet. maybe we should just merge it now, people try on next version bump and update if incorrect

@rclark binary docs are ticketed in https://github.com/mapbox/ecs-watchbot/issues/255 and should become a new PR (in the same doc i think)

File: docs/development.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+### Releasing a version
+
+Ideally you track the release of a new version in an issue with a checklist
+
+- [ ] Merge all PRs intended for the release
+- [ ] Verify that the [changelog](../changelog.md) includes all relevant changes under the new version number
+- [ ] Determine if the new version is a patch, minor or major version bump after [SEMVER](https://semver.org/)
+- [ ] Run `npm version [patch | minor | major]`
+  - This will run the tests, bump version, update fixtures, create a commit and tag
```
Comment (arunasank) [inline docs/development.md]: @freenerd this bumps the version, creates a commit and tag with the version. We would still need to retain the instructions for updating the fixtures and update them separately.

Comment (arunasank) [reply docs/development.md]: Ah, saw that you added it to the `scripts` section of the package.json as well. This is awesome! 🙌 Please ignore the above comment.

File: docs/development.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
+## Development
+
+This guide specifies how to develop Watchbot itself and is not intended for users of Watchbot.
+
+### Tests
+
+```sh
+# populate your environment with AWS credentials
+npm test
```
Comment (jakepruitt) [inline docs/development.md]: From the look of this line, it makes it seems like `npm test` adds the credentials. Should we instead be more specific about that imperative? Like

```
# Before running tests, please populate your environment with AWS credentials, then run
npm test
```

File: docs/development.md (hunk 2 above)
Comment (jakepruitt) [reply docs/development.md]: This is so wonderful

**Comment by freenerd** [comment]
Meanwhile #306 shipped. Closing as stale.

**Pull request closed without merging**